CONTRIBUTING.md

```diff
@@ -49,6 +49,22 @@ close your PR in order to prevent work being done on it.
 [issues]: https://github.com/rust-windowing/winit/issues
 [matrix]: https://matrix.to/#/#rust-windowing:matrix.org
 
+### Formatting
+
+Winit relies on nightly `rustfmt`. A PR will usually not get reviewed if the
+nightly `cargo-fmt` check did not pass. If you have `rustup` installed you can
+install the nightly toolchain using `rustup toolchain install nightly`.
+
+Then to format the code you can run `cargo +nightly fmt`.
+
+For more on how to manage toolchains, see [the documentation](https://rust-lang.github.io/rustup/concepts/toolchains.html).
+
+If you feel uncomfortable using a `nightly` toolchain, ensure that CI passes
+with the exception of formatting, and ask the maintainers on your PR to format
+the code before merge.
```

Comment:
I don't want to encode this into a kind of rule, I think we are capable enough to make judgements like this individually.

I'm also not sure why anybody would be "uncomfortable" using a nightly toolchain, considering its just for formatting. Installing the nightly toolchain does not force you to use or ship compiled binaries with nightly.

Reply:
I'm afraid that survivalship bias does not always work, sir.